core/blockchain.go

```diff
@@ -114,6 +115,32 @@ type BlockChain struct {
 	atxi *AtxiT
 }
 
+type ChainInsertResult struct {
```

Comment:
Here's those new dedicated return types for `*Insert` methods I mentioned above. Notice that they wrap existing `Event` types. Not sure if that's a clever and resourceful way to do it, or just plain convenient, or dumb for some reason I don't see. 

👓 PTAL.